usl_models/usl_models/flood_ml/model.py

```diff
@@ -118,12 +173,82 @@ def forward(
 
         # Concatenate and feed into remaining ConvLSTM and TransposeConv layers
         # [B, n, H', W', k'] -> [B, H, W, 1]
-        lstm_input = self.concat([st_cnn_output, geo_cnn_output, temp_input])
+        lstm_input = tf.concat([st_cnn_output, geo_cnn_output, temp_input], axis=-1)
         lstm_output = self.conv_lstm(lstm_input)
         output = self.output_cnn(lstm_output)
 
         return output
 
-    def call(self):
-        """Runs the entire autoregressive model."""
-        raise NotImplementedError
+    def call(self, inputs):
+        """Runs the entire autoregressive model.
+
+        Args:
+            inputs: A dictionary of input tensors.
+                - "geospatial" (required): [H, W, f]
+                - "temporal" (required): [T]
+                - "spatiotemporal" (optional): a single flood map [H, W]
+
+        Returns:
+            A tuple (all_flood_predictions, max_flood_map) of the flood predictions and
+            the max predicted flood depth at each cell over the entire storm duration.
+            The maps are the same shape as the input: (H, W).
+        """
+        # TODO: Add data validation
+        geo_input = inputs.get("geospatial")
```

Comment:
It looks like this will fail below if 'geospatial' isn't present in the dictionary and this returns a None. I'd expect MyPy to return a type errors below because the code doesn't actually support `geo_input` being None. Some thoughts:
can do
```python
inputs["geospatial"]
```
to just go ahead and get the error rather than propagate it forward into unhandled Nones

but if 'geospatial' is in fact a required argument, we can reflect that in the signature for this function 
```python
def call(self, geospatial: Tensor, temporal: Tensor, spatiotemporal: Optional[Tensor]):
```
If you'd rather accept a single input object rather than something like the above, I'd encode the requirements in a data structure rather than a genric dictionary ala
```python
@dataclasses.dataclass(slots=True)
class ModelInputs:
    geospatial: Tensor
    temporal: Tensor
    spatiotemporal: Optional[Tensor] = None

def call(self, inputs: ModelInputs):
```

Reply:
The `call` method expects a dict or list of tensors, so I'm not sure how much flexibility we have in defining a data structure; I'd have to test it out.

We do require geospatial and temporal inputs to always be passed. I currently have a TODO to add some data validation in the future, maybe to be done in a preprocessing step, e.g., with feature scaling. But maybe removing the `get` and calling `inputs["geospatial"]` is a sufficient first step here.

Reply:
Ah, I haven't used TensorFlow in quite some time, some I'm not familiar with the class-based `call` interface. It looks kinda sk-learn-y?
So in that case, I'd just add something like
```python
if 'geospatial' not in inputs:
    raise ValueError('Missing required tensor "geospatial"')
```
and ignore everything else I wrote :)

Reply:
Done. Pulled this out into a separate preprocessing function.